index.d.ts

```diff
@@ -632,6 +632,15 @@ declare namespace i18next {
       key: TKeys | TKeys[],
       options?: TOptions<TInterpolationMap> | string,
     ): TResult;
+    <
+      TResult extends string | object | Array<string | object> | undefined | null = string,
+      TKeys extends string | TemplateStringsArray = string,
+      TInterpolationMap extends object = StringMap
+    >(
```

Comment:
@rosskevin I'm not really experienced enough in TypeScript so that I have to define the generics in TFunction only once. Do you know how to do that?

Reply:
I was able to get a little less cruft/more reuse, but was unable to declare types at the interface level.  It's a bit better but very much the same as you authored.